### DISCUSSION TITLE
btn on/off state unclear with

### DISCUSSION THREAD

**Discussion by erwin** [HASwitchPlate/openHASP#329] [Q&A] [answered]
When using any theme besides `Mono`, setting my button `btn` to on/off isn't visually distinguishable:

Button `青色` (Cyan) activated:

![image](https://user-images.githubusercontent.com/11534217/166206655-6d1a41d3-3768-46e3-a65d-547dedabb485.png)

Button `红色` (Red) activated, Cyan deactivated:

![image](https://user-images.githubusercontent.com/11534217/166206615-41f25564-89e7-494a-b022-b7b553edce33.png)

I can't visually deduce which state the button is at.

When I change to Mono, it's not very attactive, but I can see which button is active (has a property > val set to 1)

青色 Cyan
![image](https://user-images.githubusercontent.com/11534217/166207243-a6d962c6-755c-4abf-b793-540fd80e283c.png)

红色 Red

![image](https://user-images.githubusercontent.com/11534217/166207283-54dfd976-1e2e-4897-ac7f-355c3fe43b5c.png)

彩虹 Rainbow

![image](https://user-images.githubusercontent.com/11534217/166207320-70e95276-7b34-4c71-b361-bda3833add0c.png)

Any tips for how to better design my button layout so that which button is active will be more clear when using my favorite `Hasp Dark` theme?

**Comment by fvanroie** [comment, marked as answer]
I think the darker border you see in the mono theme is not an indication of the on/off state, but a focus indicator of the last touched object.

What are you trying to achieve exactly? If you want to have toggle buttons, use `"toggle":1`. If you want to have only one selected button active, use `btnmatrix` with `"toggle":1,"one_check":1`.

**Comment by erwin** [comment]
That's what I've been looking for! Much better... Hadn't paid attention to the `btn` `toggle:` property and was even setting it to false (the default) in many places...

https://openhasp.haswitchplate.com/0.6.3/design/objects/#button

![image](https://user-images.githubusercontent.com/11534217/166615094-debc7f02-0a73-4698-9dc2-267efe5f58c6.png)

Now the toggled state of my 3 color selection buttons works perfectly!

```
{"page":  5, "id":  60, "x":  8, "y":240, "w":  68, "h": 30, "obj": "btn", "text":"青色", "toggle":1, "align":1, "text_font":"chinese_24"},
{"page":  5, "id":  61, "x": 84, "y":240, "w":  68, "h": 30, "obj": "btn", "text":"红色", "toggle":1, "align":1, "text_font":"chinese_24"},
{"page":  5, "id":  62, "x":164, "y":240, "w":  68, "h": 30, "obj": "btn", "text":"彩虹", "toggle":1, "align":1, "text_font":"chinese_24"},

```
